src/__tests__/index.js

```diff
@@ -1,63 +1,86 @@
+    expect(init).toThrowError('Analytics is not logging! You must initialize analytics with the correct params.');
   });
-  it('default is a function', async () => {
-    OKGAnalytics.default.should.be.a('function');
+
+  test('Throws when analytics is missing', () => {
+    const init = () => OKGAnalytics({});
+    expect(init).toThrowError('Analytics is not logging! You must initialize analytics with the correct params.');
   });
 
-  it('analytics is an object on window', async () => {
-    // eslint-disable-next-line no-undef
-    window.analytics.should.be.a('object');
+  test('Throws when integrations is missing', () => {
+    const init = () => OKGAnalytics({ analytics });
+    expect(init).toThrowError('Analytics is not logging! You must initialize analytics with the correct params.');
   });
 
-  it('window.history.pushState is function', async () => {
-    // eslint-disable-next-line no-undef
-    window.history.pushState.should.be.a('function');
+  test('Throws when analytics is missing all core functions', () => {
+    const missingAll = () => OKGAnalytics({ analytics: {}, integrations: {} });
+    expect(missingAll).toThrowError('Analytics is not logging! Expected analytics to contain ready ,track ,page ,identify , function(s).');
   });
 
-  it('window.onload is function', async () => {
-    const settings = {
-      'Google Analytics': { 'trackingId': 'UA-58359748-3' }, // eslint-disable-line quote-props
-      'Mixpanel': { 'token': 'b513b13a2e253576934b47d2a195ae29', 'people': true }, // eslint-disable-line quote-props
-    };
+  test('Throws when analytics is missing ready()', () => {
+    const missingReady = () => OKGAnalytics({ analytics: { track, page, identify }, integrations: {} });
+    expect(missingReady).toThrowError('Analytics is not logging! Expected analytics to contain ready , function(s).');
+  });
 
-    // eslint-disable-next-line no-undef
-    should.not.exist(window.onload);
+  test('Throws when analytics is missing track()', () => {
+    const missingTrack = () => OKGAnalytics({ analytics: { ready, page, identify }, integrations: {} });
+    expect(missingTrack).toThrowError('Analytics is not logging! Expected analytics to contain track , function(s).');
+  });
 
-    OKGAnalytics.default(settings);
+  test('Throws when analytics is missing page()', () => {
+    const missingPage = () => OKGAnalytics({ analytics: { ready, track, identify }, integrations: {} });
+    expect(missingPage).toThrowError('Analytics is not logging! Expected analytics to contain page , function(s).');
+  });
 
-    // eslint-disable-next-line no-undef
-    window.onload.should.be.a('function');
+  test('Throws when analytics is missing identify()', () => {
+    const missingIdentify = () => OKGAnalytics({ analytics: { ready, track, page }, integrations: {} });
+    expect(missingIdentify).toThrowError('Analytics is not logging! Expected analytics to contain identify , function(s).');
   });
 });
 
```

Comment:
The only other test ideas I have would be if we can setup mock functions in the `analytics` object and the execute the `*WhenReady()` functions and see if those mocked functions were called.